### DISCUSSION TITLE
Equivalent of SwinjectAutoregistration

### DISCUSSION THREAD

**Discussion by MostafaNafie** [hmlongco/Factory#245] [Q&A] [answered]
Hi there,

Is there an equivalent of [SwinjectAutoregistration](https://github.com/Swinject/SwinjectAutoregistration) functionality in Factory?

Especially this example:
![Screenshot 2024-10-24 at 10 25 38 AM](https://github.com/user-attachments/assets/25ae76de-dc39-44d5-bb17-53a4f26353cc)

Thank you!

**Comment by hmlongco** [comment, marked as answer]
Swinject allows you to register and resolve based solely on on type... and a registration for a given type may not, in fact, exist. 

Factory requires a known, computed variable on a container to exist, which in turn makes it compile time safe.

So no.

**Reply by MostafaNafie** [reply]
Thank you for your answer and for your efforts working on Factory.
